pkgs/by-name/cl/clean/package.nix

```diff
@@ -1,50 +1,45 @@
 {
+  binutils,
+  fetchurl,
+  gcc,
   lib,
   stdenv,
-  fetchurl,
 }:
 
 stdenv.mkDerivation {
   pname = "clean";
-  version = "3.0";
+  version = "3.1";
 
   src =
     if stdenv.hostPlatform.system == "i686-linux" then
       (fetchurl {
-        url = "https://ftp.cs.ru.nl/Clean/Clean30/linux/clean3.0_32_boot.tar.gz";
-        sha256 = "0cjxv3vqrg6pz3aicwfdz1zyhk0q650464j3qyl0wzaikh750010";
+        url = "https://ftp.cs.ru.nl/Clean/Clean31/linux/clean3.1_32_boot.tar.gz";
+        sha256 = "Ls0IKf+o7yhRLhtSV61jzmnYukfh5x5fogHaP5ke/Ck=";
       })
     else if stdenv.hostPlatform.system == "x86_64-linux" then
       (fetchurl {
-        url = "https://ftp.cs.ru.nl/Clean/Clean30/linux/clean3.0_64_boot.tar.gz";
-        sha256 = "06k283y9adbi28f78k3m5ssg6py73qqkz3sm8dgxc89drv4krl2i";
+        url = "https://ftp.cs.ru.nl/Clean/Clean31/linux/clean3.1_64_boot.tar.gz";
+        sha256 = "Gg5CVZjrwDBtV7Vuw21Xj6Rn+qN1Mf6B3ls6r/16oBc=";
       })
     else
       throw "Architecture not supported";
 
-  hardeningDisable = [
-    "format"
-    "pic"
-  ];
-
-  # clm uses timestamps of dcl, icl, abc and o files to decide what must be rebuild
-  # and for chroot builds all of the library files will have equal timestamps.  This
-  # makes clm try to rebuild the library modules (and fail due to absence of write permission
-  # on the Nix store) every time any file is compiled.
```

Comment:
I'd like to keep these comments in if possible, the magic that we need to do for Clean is a little strange.

Reply:
I thought that this was already explained in the header of the corresponding patch. I'd prefer to keep the comment as close to the patch as possible, so I merged it into the header. Better that way or do you rather prefer the comment in the derivation?

Reply:
No, you're right. That's good!